### PR TITLE
fix transaction error of synced Realm.

### PR DESCRIPTION
**Pull request by zaki50**

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -422,6 +422,8 @@ private static void initializeSyncedRealm(Realm realm) {
         } finally {
             if (commitChanges) {
                 realm.commitTransaction(false);
+            } else {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Also in `initializeRealm`?

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Nvm, it does it already

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -422,6 +422,8 @@ private static void initializeSyncedRealm(Realm realm) {
...
+                realm.cancelTransaction();
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: I think it's safe to check first if we're in a transaction `realm.isInTransaction()`

Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: it should never happens ... if i happens, then it is a serious bug ...

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: The transaction is started when you enter the method?

**Pull request merged**